File: lib/Counter/State.js
import List from "../fable-core/List";
export function init() {
  return [0, new List()];
}
export function update(msg, model) {
  if (msg.tag === 1) {
    return [model - 1, new List()];
  } else if (msg.tag === 2) {
    return [0, new List()];
  } else {
    return [model + 1, new List()];
  }
}

File: src/Home/State.fs
module Home.State

open Elmish
open Types

let init () : Model * Cmd<Msg> =
  {input = ""; lazyView = ""}, []

let update msg model : Model * Cmd<Msg> =
  printfn "update"
  match msg with
  | ChangeStr str ->
      {model with input=str}, []
  | ChangeLazyView str ->
      {model with lazyView=str}, []
